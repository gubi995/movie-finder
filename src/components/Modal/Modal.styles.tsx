import styled from '@emotion/styled';
import { Modal, ButtonBase } from '@mui/material';

import { BACKGROUND } from '@/styles/linear-gradients';

export const StyledModal = styled(Modal)`
  display: grid;
  place-items: center;
`;

export const ModalContent = styled.div`
  display: flex;
  position: relative;
  max-width: min(50vw, 700px);
  min-width: 250px;
  max-height: 80vh;
  background-color: white;
  border-radius: 3px;
  padding: 3rem;
`;

export const CloseButton = styled(ButtonBase)`
  position: absolute;
  right: 0;
  top: 0;
  transform: translate(20%, -20%);
  background: ${BACKGROUND};
  color: white;
  width: 30px;
  height: 30px;
  border-radius: 50%;
`;
