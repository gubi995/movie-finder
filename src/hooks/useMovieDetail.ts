import { useQuery } from 'react-query';

import { MovieDetail } from '@/model/movie';
import { fetchWikipediaPage } from '@/api/wikipedia';

export const useMovieDetail = (title: string) =>
  useQuery<MovieDetail, Error>(['movieDetail', title], () => fetchWikipediaPage(title), {
    refetchOnWindowFocus: false,
  });
