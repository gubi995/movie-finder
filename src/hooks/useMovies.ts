import { useQuery } from 'react-query';

import { searchMovies } from '@/api/movies';
import { Movie } from '@/model/movie';

export const useMovies = (title: string) =>
  useQuery<Movie[], Error>(['movie', title], () => {
    if (title) {
      return searchMovies(title);
    }

    return [];
  });
