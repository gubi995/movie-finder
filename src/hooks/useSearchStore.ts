import create from 'zustand';

interface State {
  term: string;
  updateTerm: (term: string) => void;
}

export const useSearchStore = create<State>((set) => ({
  term: '',
  updateTerm: (term: string) => set({ term }),
}));
