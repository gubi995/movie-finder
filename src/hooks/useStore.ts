import { ReactNode } from 'react';
import create from 'zustand';

interface State {
  searchTerm: string;
  modalContent: ReactNode;
  updateSearchTerm: (term: string) => void;
  setModalContent: (term: ReactNode) => void;
}

export const useStore = create<State>((set) => ({
  searchTerm: '',
  modalContent: null,
  updateSearchTerm: (term: string) => set({ searchTerm: term }),
  setModalContent: (content: ReactNode) => set({ modalContent: content }),
}));
