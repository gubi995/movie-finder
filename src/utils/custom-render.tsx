import { render, RenderOptions } from '@testing-library/react';
import { FC, ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

const AllTheProviders: FC = ({ children }) => {
  return <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>;
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';

export { queryClient };

export { customRender as render };
