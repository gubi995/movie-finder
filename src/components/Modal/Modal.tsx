import { Close } from '@mui/icons-material';

import { useStore } from '@/hooks/useStore';

import { CloseButton, ModalContent, StyledModal } from './Modal.styles';

const Modal = () => {
  const { modalContent, setModalContent } = useStore(({ modalContent, setModalContent }) => ({
    modalContent,
    setModalContent,
  }));

  const closeModal = () => setModalContent(null);

  return (
    <StyledModal open={!!modalContent} onClose={closeModal}>
      <ModalContent>
        <CloseButton onClick={closeModal}>
          <Close />
        </CloseButton>
        {modalContent}
      </ModalContent>
    </StyledModal>
  );
};

export default Modal;
