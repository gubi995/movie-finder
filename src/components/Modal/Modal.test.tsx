import userEvent from '@testing-library/user-event';

import { screen, render, cleanup } from '@/utils/custom-render';
import * as hooks from '@/hooks/useStore';

import Modal from './Modal';

const modalMock = { modalContent: <div>Modal content</div>, setModalContent: jest.fn() };

jest.spyOn(hooks, 'useStore').mockReturnValue(modalMock);

describe('Modal component', () => {
  beforeEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test('should render with content', () => {
    const { baseElement } = render(<Modal />);

    expect(baseElement).toMatchSnapshot();
  });

  test('should hide modal on close button click', () => {
    render(<Modal />);

    const closeButton = screen.getByRole('button');

    userEvent.click(closeButton);

    expect(modalMock.setModalContent).toHaveBeenCalledTimes(1);
    expect(modalMock.setModalContent).toHaveBeenCalledWith(null);
  });
});
