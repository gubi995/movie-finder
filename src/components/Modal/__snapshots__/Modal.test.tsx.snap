// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render with content 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  display: grid;
  place-items: center;
}

.emotion-1 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  max-width: min(50vw, 700px);
  min-width: 250px;
  background-color: white;
  border-radius: 3px;
  padding: 3rem;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transform: translate(20%, -20%);
  -moz-transform: translate(20%, -20%);
  -ms-transform: translate(20%, -20%);
  transform: translate(20%, -20%);
  background: linear-gradient(90deg, #00d2ff 0%, #3a47d5 100%);
  color: white;
  width: 30px;
  height: 30px;
  border-radius: 50%;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root emotion-0"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root emotion-1"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="emotion-2"
      tabindex="-1"
    >
      <button
        class="MuiButtonBase-root emotion-3"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-4"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root emotion-5"
        />
      </button>
      <div>
        Modal content
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
